src/components/fields/MultiColorPicker.js

```diff
@@ -0,0 +1,166 @@
+import React, {Component} from 'react';
+import Color from './Color';
+import Colorscale from './Colorscale';
+import Info from './Info';
+import Field from './Field';
+import RadioBlocks from '../widgets/RadioBlocks';
+import PropTypes from 'prop-types';
+import {adjustColorscale, connectToContainer} from 'lib';
+
+class UnconnectedMultiColorPicker extends Component {
+  constructor(props, context) {
+    super(props, context);
+    this.state = {
+      selectedConstantColorOption:
+        context.traceIndexes.length > 1 &&
+        props.fullValue.every(v => v[1] === props.fullValue[0][1])
+          ? 'single'
+          : 'multiple',
+    };
+    this.setColor = this.setColor.bind(this);
+    this.setColors = this.setColors.bind(this);
+  }
+
+  setColor(color) {
+    this.props.updatePlot(color);
+  }
+
+  setColors(colorscale, colorscaleType) {
+    const numberOfTraces = this.context.traceIndexes.length;
+    const colors = colorscale.map(c => c[1]);
+
+    let adjustedColors = colors;
+
+    if (colorscaleType !== 'categorical') {
+      adjustedColors = adjustColorscale(colors, numberOfTraces, colorscaleType);
+    }
+
+    if (
+      adjustedColors.every(c => c === adjustedColors[0]) ||
+      colorscaleType === 'categorical'
+    ) {
+      adjustedColors = adjustColorscale(
+        colors,
+        numberOfTraces,
+        colorscaleType,
+        {repeat: true}
+      );
+    }
+
+    const updates = adjustedColors.map(color => ({
+      [this.props.attr]: color,
+    }));
+
+    this.context.updateContainer(updates);
+  }
+
+  render() {
+    const _ = this.context.localize;
+    const constantOptions = [
+      {label: _('Single'), value: 'single'},
+      {label: _('Multiple'), value: 'multiple'},
+    ];
+    const selectedConstantColorOption = this.props.parentState
+      ? this.props.parentState.selectedConstantColorOption
```

Comment:
parentState only has one variable, why not just have parentSelectedConstantColorOptions?

Reply:
I'm using parent state to pass MarkerColor state, it has more than one variable.

Reply:
I'm not seeing other variables